### PR TITLE
stdenv: bootstrap cmake and python on darwin

### PR DESCRIPTION
**Pull request by LnL7**
###### Motivation for this change

Use bootstrap versions of cmake and python to get some dependencies like curl, libarchive and sqlite out of the stdenv.

/cc @copumpkin 

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change: [lnl7-wip](http://hydra.nixos.org/eval/1318255?compare=trunk)
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

```diff
-CF-osx-10.9.5-private.drv
-DevSDK_OSX109.pkg.drv
-MacOS_SDK-10.9.drv
-OSXPrivateSDK-f4d52b60e86b496abfaffa119a7d299562d99783-src.drv
-apple-framework-Accelerate.drv
-apple-framework-ApplicationServices.drv
-apple-framework-CFNetwork.drv
-apple-framework-CoreAudio.drv
-apple-framework-CoreData.drv
-apple-framework-CoreGraphics.drv
-apple-framework-CoreServices.drv
-apple-framework-CoreText.drv
-apple-framework-CoreWLAN.drv
-apple-framework-DiskArbitration.drv
-apple-framework-IOBluetooth.drv
-apple-framework-IOKit.drv
-apple-framework-IOSurface.drv
-apple-framework-ImageIO.drv
-apple-framework-NetFS.drv
-apple-framework-OpenDirectory.drv
-apple-framework-Security.drv
-apple-framework-SecurityFoundation.drv
-apple-framework-ServiceManagement.drv
-apple-framework-SystemConfiguration.drv
-apple-lib-xpc.drv
-cmake-3.7.1.drv
+cmake-boot-3.7.1.drv
-curl-7.51.0.drv
-db-5.3.28.drv
-expat-2.2.0.drv
-gdbm-1.12.drv
-gdbm-1.12.drv
-libarchive-3.2.2.drv
-libssh2-1.8.0.drv
-libuv-1.10.1.drv
-libuv-v1.10.1-src.drv
-lzo-2.09.drv
-openssl-1.0.2j.drv
-openssl-1.0.2j.drv
-python-2.7.13.drv
+python-boot-2.7.12.drv
-readline-6.3p08.drv
-sharutils-4.11.1.drv
-sqlite-3.15.2.drv
-xar-1.5.2.drv
```

### CODE REVIEW

File: pkgs/development/interpreters/python/cpython/2.7/boot.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      is used in a wide variety of application domains. Some of its key
+      distinguishing features include: clear, readable syntax; strong
+      introspection capabilities; intuitive object orientation; natural
+      expression of procedural code; full modularity, supporting
+      hierarchical packages; exception-based error handling; and very
+      high level dynamic data types.
+    '';
+    license = stdenv.lib.licenses.psfl;
+    platforms = stdenv.lib.platforms.all;
+    maintainers = with stdenv.lib.maintainers; [ lnl7 chaoflow domenkozar ];
```
Comment (FRidh) [inline pkgs/development/interpreters/python/cpython/2.7/boot.nix]: I guess you or darwin maintainers are going to maintain this expression?

**Review by FRidh — APPROVED** [review]

File: pkgs/development/interpreters/python/cpython/2.7/boot.nix (hunk 1 above)
Comment (LnL7) [inline pkgs/development/interpreters/python/cpython/2.7/boot.nix]: Yes, I separated this so changes in the default python expression won't necessarily cause a rebuild of the entire darwin world.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -5176,9 +5176,13 @@ in
     inherit (stdenvAdapters) overrideCC;
   };
 
-  llvmPackages_37 = callPackage ../development/compilers/llvm/3.7 {
+  llvmPackages_37 = callPackage ../development/compilers/llvm/3.7 ({
```
Comment (LnL7) [inline pkgs/top-level/all-packages.nix]: I'm pretty sure I should do this somewhere else but I couldn't get it working in the stdenv bootstrap phases.

**Comment by LnL7** [comment]
Merging straight to master because of https://github.com/NixOS/nixpkgs/commit/125fa1520dd5bd3255badd472cd9ce70630dc67f

**Pull request merged**